Plugins/DeleteUnusedBranches/DeleteUnusedBranchesForm.cs

```diff
@@ -125,21 +127,26 @@ private static IEnumerable<string> GetObsoleteBranchNames(RefreshContext context
 
             var args = new GitArgumentBuilder("branch")
             {
+                 "--list",
                  { context.IncludeRemotes, "-r" },
                  { !context.IncludeUnmerged, "--merged" },
                  context.ReferenceBranch
             };
 
-            string[] branches = context.Commands.GitExecutable.GetOutput(args)
-                .Split('\n')
-                .Where(branchName => !string.IsNullOrEmpty(branchName))
-                .Select(branchName => branchName.Trim('*', ' ', '\n', '\r'))
-                .Where(branchName => branchName != "HEAD" && branchName != curBranch &&
-                                   branchName != context.ReferenceBranch)
-                .ToArray();
+            var result = context.Commands.GitExecutable.Execute(args);
 
-            return branches.Where(branchName => (!context.IncludeRemotes || branchName.StartsWith(context.RemoteRepositoryName + "/")) &&
-                                   (regex == null || regex.IsMatch(branchName) == regexMustMatch)).ToArray();
+            if (result.ExitedSuccessfully)
+            {
+                return _commandOutputParser.GetBranchNames(result.AllOutput)
+                                           .Where(branchName => branchName != curBranch && branchName != context.ReferenceBranch)
+                                           .Where(branchName => (!context.IncludeRemotes || branchName.StartsWith(context.RemoteRepositoryName + "/")) && (regex == null || regex.IsMatch(branchName) == regexMustMatch))
+                                           .ToArray();
```

Comment:
The method returns `IEnumerable<T>`, and the caller will materialise the sequence.
Why do we need to enumerate and allocate here?

Reply:
Honestly? It was materializing before and it's materializing now :)
I had a look at the caller method and you are right: that ToArray is useless. 
I'll fix it.